internal/adapters/gitlabtokenrefresher/gitlabtokenrefresher.go

```diff
@@ -0,0 +1,105 @@
+package gitlabtokenrefresher
```

Comment:
Is there a reason for making this gitlab specific?

Ideally we should have one package that refreshes all oauth tokens. Not a gitlab specific one. As long as it is oauth2 we should be able to handle it.

Reply:
the Token struct in the oauth2 package didn't fit the response from GitLab when receiving the tokens back, so I don't know if this would work with any other oauth provider. Hence naming it something gitlab-specific.

Reply:
@wesjdj please modify the struct as you see fit so we can use the same one for any oauth provider.

Here is an example response from keycloak after a refresh. Which should be the same as an initial login.
```
{
    "access_token": "XXXX",
    "expires_in": 1800,
    "refresh_expires_in": 86400,
    "refresh_token": "XXXX",
    "token_type": "Bearer",
    "not-before-policy": 0,
    "session_state": "XXXX",
    "scope": "address profile email"
}
```

The response from Gitlab is like this:
```
{
    "access_token": "XXXXXX",
    "token_type": "Bearer",
    "expires_in": 7200,
    "refresh_token": "XXXXXX",
    "scope": "api",
    "created_at": 1675947618
}
```

You can use the same struct with only the properties we want. For example something like below will work on both responses. The only difference is that in some cases the values will have their zero/default values if they are not present in the response. This is fine - because the most important parts of the response are part of the oauth standard and they will be the same even for other providers. But for now handling these two is enough and it is doable.

```go
type Token struct {
	AccessToken           string `json:"access_token"`
	RefreshToken          string `json:"refresh_token"`
	AccessTokenExpiresIn  int    `json:"expires_in"`
	RefreshTokenExpiresIn int    `json:"refresh_expires_in"`
	Scope                 string `json:"scope"`
}
```